lib/gollum/public/gollum/javascript/gollum.js.erb

```diff
@@ -482,7 +482,19 @@ $(document).ready(function() {
     $('span.critic.comment').filter(function() {return $(this).text()!="";}).before('&#8225;');
   }
 
-  $('.markdown-body p, .markdown-body span, .markdown-body pre, .markdown-body table').attr('dir','auto');
+  if($('#wiki-content').length ){
+    // Set text direction auto
+    $('.markdown-body p, .markdown-body span, .markdown-body pre, .markdown-body table').attr('dir','auto');
+
+    // Copy anchors for each editable header and give it the new anchor the 'edit' class, to display a link to edit the section
+    $('a.anchor').each(function (index, anchor) {
+      header = $(anchor).closest(':header');
+      if (header.hasClass('editable')){
+        var newUrl = routePath('edit') + '/' + pageName() + $(anchor).attr('href');
+        $(anchor).clone().addClass('edit').attr('href', newUrl).appendTo(header);
```

Comment:
Why clone the anchor here?

Reply:
Do you mean: why clone it at all, or why clone it here? Since I wanted to insert a second (edit) anchor, I thought cloning the first one might require less jQuery calls.

Reply:
I was wondering why to clone it at all; didn't realize you were adding anchors, I just thought you were modifying the existing ones.

Reply:
Needed the second anchor to display the pencil icon/edit link. Still unsure whether it's best to do that in the JS or in `gollum-lib`. As I explained in the conversation I would have preferred to do it all with AnchorJS so there is less html-generating in gollum-lib, and I thought this might be second best. If you think it may be clearer to have it all in one place (gollum-lib) let me know.

Reply:
This is fine.